### PR TITLE
Use instancing for sprite rendering

### PR DESCRIPTION
**Pull request by opstic**
# Objective

- Improve sprite rendering performance

## Solution

Store per-instance data instead of 6 vertices to the vertex buffer and change the `VertexStepMode` to `Instance`.
This allows for less data being transferred.
Originally, 1728 bits (216 bytes) had to be transferred to the GPU for each sprite.
Now, only 768 bits (96 bytes) are needed per-sprite.
### Performance
(Sprite size set to 0.01 to minimize sampler performance impact, buffer size captured from Intel GPA)
- 10000 sprites in `bevymark` 
Original: 2150408 bytes transferred per frame, 145-147 fps.
Instanced: 960004 bytes transferred per frame, 158-161 fps.

### CODE REVIEW

**Comment by mockersf** [comment]
there are FPS regressions for me on both `bevymark` and `many_sprites`:
- `many_sprites` from ~820 fps to ~760 fps
- `bevymark -- 1000 100` from 45 fps to 39 fps

**Comment by Elabajaba** [comment]
This is (surprisingly) faster on my system. (for bevymark, I'm seeing slightly better minimums, and ~112fps average vs ~90fps average for our current batching, for many_sprites perf is the same).

The old wisdom was that you want to use batching for sprites because you're drawing quads, and GPUs run wavefronts of 32-64 vertices (so you end up with very low utilization per wavefront). However, these days (some? AMD, maybe Nvidia, others idk) GPU drivers can merge instances into wavefronts. (but sometimes it merges and it's still slow? idk)

Basically, need people to test this on a bunch of different hardware and check wavefront occupancy to figure out who does merging and who doesn't and if we're okay with the performance tradeoffs. (or maybe even have both paths, but have it be an option?)

**Comment by superdump** [comment]
Thanks for the contribution. Closing this as #9597 was merged.
